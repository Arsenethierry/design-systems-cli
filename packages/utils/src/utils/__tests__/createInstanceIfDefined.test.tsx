import * as React from 'react';
import { createInstanceIfDefined } from '../createInstanceIfDefined';

test('it should create the instance', () => {
  const Text: React.FC = ({ children }) => <div>{children}</div>;

  expect(createInstanceIfDefined('A string', Text)).toMatchSnapshot();
});

test('it not create the instance', () => {
  const Text: React.FC = ({ children }) => <div>{children}</div>;

  expect(createInstanceIfDefined(null, Text)).toBeUndefined();
});
