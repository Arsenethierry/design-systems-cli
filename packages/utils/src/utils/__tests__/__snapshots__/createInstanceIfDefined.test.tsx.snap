// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should create the instance 1`] = `
<Text>
  A string
</Text>
`;
